src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -589,17 +590,19 @@ private TypeGenerationSpec ParseTypeGenerationSpec(in TypeToGenerate typeToGener
                 }
 
                 var typeRef = new TypeRef(type);
-                string typeInfoPropertyName = typeToGenerate.TypeInfoPropertyName ?? GetTypeInfoPropertyName(type);
+                string typeInfoPropertyName = typeToGenerate.TypeInfoPropertyName ??
```

Comment:
I'm not comfortable making cross-cutting changes in RC2 backports. To minimize risk, we should be making targeted changes even if it results in temporary code duplication.